src/AppBundle/Entity/LegislativeDistrictZone.php

```diff
@@ -139,11 +144,40 @@ public function getAreaCode(): ?string
 
     public function setAreaCode(string $code): void
     {
-        $this->areaCode = $code;
+        $this->areaCode = sprintf('%04s', $code);
+
+        if (null === $this->rank) {
+            $this->rank = self::guessRankFromAreaCode($code);
+        }
+    }
+
+    private static function guessRankFromAreaCode(string $areaCode): int
+    {
+        if (self::isCorsica($areaCode)) {
+            return 'A' === substr($areaCode, -1) ? 20 : 21;
+        }
+
+        $rank = (int) ltrim($areaCode, '0');
```

Comment:
I always use `(int)` instead of `intval()` function. It's an explicit casting. But I don't know if one is better than the other.